src/ui/public/agg_response/tabify/tabify.js

```diff
@@ -21,6 +21,17 @@ import _ from 'lodash';
 import { TabbedAggResponseWriter } from './_response_writer';
 import { TabifyBuckets } from './_buckets';
 
+/**
+ * Sets up the ResponseWriter and kicks off bucket collection.
+ *
+ * @param {AggConfigs} aggs - the agg configs object to which the aggregation response correlates
+ * @param {Object} esResponse - response that came back from Elasticsearch
+ * @param {Object} respOpts - options object for the ResponseWriter with params set by Courier
+ * @param {boolean} respOpts.isHierarchical - reflects the value of vis.isHierarchical()
+ * @param {boolean} respOpts.metricsAtAllLevels - reflects the value of vis.params.showMetricsAtAllLevels
+ * @param {boolean} respOpts.partialRows - reflects the value of vis.params.showPartialRows
+ * @param {Object} respOpts.timeRange - time range object, if provided
+ */
```

Comment:
Since params are being thrown around willy-nilly and hard to track, I added this to reduce confusion for people working on Tabify in the future. (A TypeScript rewrite would be very helpful here).